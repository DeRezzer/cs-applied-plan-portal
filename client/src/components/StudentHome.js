import React from "react";
import NavBar from "./Navbar";
import {getToken, getProfile} from "../utils/authService";

export default class StudentHome extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      plans: null
    };

    this.getAllPlans = this.getAllPlans.bind(this);
    this.renderStatus = this.renderStatus.bind(this);

    this.getAllPlans();
  }

  async getAllPlans() {
    const profile = getProfile();
    const token = getToken();
    const server = `${process.env.REACT_APP_API_HOST}:${process.env.REACT_APP_API_PORT}`;
    const getUrl = `http://${server}/user/${profile.sub}/plans/` +
      `?accessToken=${token}`;
    console.log(getUrl);
    let obj = [];

    try {
      const results = await fetch(getUrl);
      if (results.ok) {
        obj = await results.json();
        this.setState({
          plans: obj
        });
      } else {
        // we got a bad status code
        try {
          throw results;
        } catch (err) {
          err.text().then(errorMessage => {
            alert("Error " + errorMessage);
          });
        }
      }
    } catch (err) {
      alert(err);
    }
  }

  renderStatus(status) {
    switch (status) {
      case 0:
        return "Rejected";
      case 1:
        return "Awaiting student changes";
      case 2:
        return "Awaiting review";
      case 3:
        return "Awaiting final review";
      case 4:
        return "Accepted";
      default:
        return "Undefined status";
    }
  }

  goToPlan(plan) {
    window.location.href = `/viewPlan/${plan.planId}`;
  }

  render() {
    return (
      <div>
        <NavBar />
        <table className="student-plans-table">
          <thead>
            <tr>
              <th className="student-plans-data">Name</th>
              <th className="student-plans-data">Status</th>
              <th className="student-plans-data">Updated</th>
              <th className="student-plans-data">Reviewers</th>
            </tr>
            {this.state.plans ? this.state.plans.map(p =>
              <tr key={p.planId + "a"} onClick={() => this.goToPlan(p)}>
                <td className="student-plans-data" key={p.planId + "b"}>{p.planName}</td>
                <td className="student-plans-data" key={p.planId + "c"}>{this.renderStatus(p.status)}</td>
                <td className="student-plans-data" key={p.planId + "d"}>{p.lastUpdated}</td>
              </tr>) : null}
          </thead>
        </table>
        <button className="new-plan-button" onClick={() => window.location.href = "/createPlan"}>+</button>
      </div>
    );
  }
}