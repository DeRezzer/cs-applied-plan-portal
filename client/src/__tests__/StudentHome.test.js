/* eslint-disable no-undef */
import React from "react";
import Enzyme, {mount} from "enzyme";
import {BrowserRouter as Router} from "react-router-dom";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import Cookies from "js-cookie";

import setCookies from "../cookieInfo";
import StudentHome from "../components/student_home/StudentHome";


Enzyme.configure({adapter: new Adapter()});

describe("Student home", () => {

  beforeEach(() => {
    // create a mock function using jest.fn()
    const mockSet = jest.fn();

    // here we are trying to mock the 'set' functionality of Cookie
    Cookies.set = mockSet;

    // call the set method of Cookies
    setCookies("userId", "12002489701"); // Luke Skywalker
    setCookies("role", "1");

  });

  it("matches snapshot", () => {
    const home = renderer.create(
      <Router>
        <StudentHome />
      </Router>).toJSON();
    expect(home).toMatchSnapshot();
  });

  it("allows user to view a plan", () => {
    const home = renderer.create(
      <Router>
        <StudentHome />
      </Router>);
    let tree = home.toJSON();
    expect(tree).toMatchSnapshot();

    const instance = mount(
      <Router>
        <StudentHome />
      </Router>);
    instance.find("tr[key=19]").simulate("click");
    tree = instance.toJSON();
    expect(tree).toMatchSnapshot();
  });
});