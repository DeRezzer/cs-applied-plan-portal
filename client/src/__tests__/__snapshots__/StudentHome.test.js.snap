// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Student home allows user to view a plan 1`] = `
<div
  className="css-1t10nat-StudentHome"
  id="student-home-page"
>
  <div
    className="loader-container css-3s2go1-PageSpinner"
  >
    <div
      className="css-jv9gnm"
    >
      <div
        className="css-bmodww"
      />
      <div
        className="css-phb2pc"
      />
    </div>
  </div>
  <div
    className="navbar-parent css-12g91iz-Navbar"
    id="navbar"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <p
        className="osu-logo"
      >
        OSU CS Applied Plan Portal
      </p>
    </a>
    <div
      className="right-container"
    >
      <div
        className="history-dropdown css-en487y-History"
      >
        <button
          className="drop-button"
        >
          History
          <i
            className="fa fa-caret-down"
          />
        </button>
        <div
          className="dropdown-content"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <p>
              No recently visited plans
            </p>
          </a>
        </div>
      </div>
      <div
        className="notification-dropdown css-3i4jcj-Notifications"
      >
        <button
          className="drop-button-notification"
          data-count={0}
        >
          Notifications
          <span
            className="badge"
          />
          <i
            className="fa fa-caret-down"
          />
        </button>
        <div
          className="dropdown-content"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <p>
              No new notifications
            </p>
          </a>
        </div>
      </div>
      <button
        className="logout-button css-1k1xu2y-Logout"
        onClick={[Function]}
      >
        Log Out
      </button>
    </div>
  </div>
  <div
    id="student-home-container"
  >
    <div
      id="student-home-contents-container"
    >
      <div
        className="empty-plan-container"
      >
        <h3
          className="empty-plan-title"
        >
          You haven't created any plans.
        </h3>
        <a
          className="empty-plan-create-button"
          href="createPlan"
          title="Create a plan"
        >
          Create a plan
        </a>
      </div>
      <a
        className="new-plan-button"
        href="createPlan"
        title="Create a plan"
      />
    </div>
  </div>
</div>
`;

exports[`Student home matches snapshot 1`] = `
<div
  className="css-1t10nat-StudentHome"
  id="student-home-page"
>
  <div
    className="loader-container css-3s2go1-PageSpinner"
  >
    <div
      className="css-jv9gnm"
    >
      <div
        className="css-bmodww"
      />
      <div
        className="css-phb2pc"
      />
    </div>
  </div>
  <div
    className="navbar-parent css-12g91iz-Navbar"
    id="navbar"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <p
        className="osu-logo"
      >
        OSU CS Applied Plan Portal
      </p>
    </a>
    <div
      className="right-container"
    >
      <div
        className="notification-dropdown css-3i4jcj-Notifications"
      >
        <button
          className="drop-button-notification"
          data-count={0}
        >
          Notifications
          <span
            className="badge"
          />
          <i
            className="fa fa-caret-down"
          />
        </button>
        <div
          className="dropdown-content"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <p>
              No new notifications
            </p>
          </a>
        </div>
      </div>
      <button
        className="logout-button css-1k1xu2y-Logout"
        onClick={[Function]}
      >
        Log Out
      </button>
    </div>
  </div>
  <div
    id="student-home-container"
  >
    <div
      id="student-home-contents-container"
    >
      <div
        className="empty-plan-container"
      >
        <h3
          className="empty-plan-title"
        >
          You haven't created any plans.
        </h3>
        <a
          className="empty-plan-create-button"
          href="createPlan"
          title="Create a plan"
        >
          Create a plan
        </a>
      </div>
      <a
        className="new-plan-button"
        href="createPlan"
        title="Create a plan"
      />
    </div>
  </div>
</div>
`;
